pkg/oc/bootstrap/clusterup/components/service-catalog/service_catalog.go

```diff
@@ -0,0 +1,149 @@
+	ServiceCatalogServiceIP = "172.30.1.2"
+)
+
+type ServiceCatalogComponentOptions struct {
+	OCImage         string
+	MasterConfigDir string
+	ImageFormat     string
+	PublicMasterURL string
+}
+
+func (c *ServiceCatalogComponentOptions) Name() string {
+	return "openshift-service-catalog"
+}
+
+func (c *ServiceCatalogComponentOptions) Install(dockerClient dockerhelper.Interface, logdir string) error {
+	clusterAdminKubeConfigBytes, err := ioutil.ReadFile(path.Join(c.MasterConfigDir, "admin.kubeconfig"))
+	if err != nil {
+		return err
+	}
+	restConfig, err := kclientcmd.RESTConfigFromKubeConfig(clusterAdminKubeConfigBytes)
+	if err != nil {
+		return err
+	}
+	kubeClient, err := kclientset.NewForConfig(restConfig)
+	if err != nil {
+		return errors.NewError("cannot obtain API clients").WithCause(err)
+	}
+	kubeExternalClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return errors.NewError("cannot obtain API clients").WithCause(err)
+	}
+
+	for _, role := range getServiceCatalogClusterRoles() {
+		if _, err := (&reconciliation.ReconcileRoleOptions{
+			Confirm:                true,
+			RemoveExtraPermissions: false,
+			Role: reconciliation.ClusterRoleRuleOwner{ClusterRole: &role},
+			Client: reconciliation.ClusterRoleModifier{
+				Client: kubeClient.Rbac().ClusterRoles(),
+			},
+		}).Run(); err != nil {
+			return errors.NewError("could not reconcile service catalog cluster role %s", role.Name).WithCause(err)
+		}
+	}
+
+	// Instantiate service catalog
+	imageTemplate := variable.NewDefaultImageTemplate()
+	imageTemplate.Format = c.ImageFormat
+	imageTemplate.Latest = false
+
+	params := map[string]string{
+		"SERVICE_CATALOG_SERVICE_IP": ServiceCatalogServiceIP,
+		"CORS_ALLOWED_ORIGIN":        c.PublicMasterURL,
+		"SERVICE_CATALOG_IMAGE":      imageTemplate.ExpandOrDie("service-catalog"),
+	}
+
+	// Stands up the service catalog apiserver, etcd, and controller manager
+	component := componentinstall.Template{
+		Name:            "service-catalog",
+		Namespace:       catalogNamespace,
+		RBACTemplate:    bootstrap.MustAsset("examples/service-catalog/service-catalog-rbac.yaml"),
+		InstallTemplate: bootstrap.MustAsset("examples/service-catalog/service-catalog.yaml"),
+
+		// Wait for the apiserver endpoint to become available
+		WaitCondition: func() (bool, error) {
+			glog.V(2).Infof("polling for service catalog api server endpoint availability")
+			deployment, err := kubeExternalClient.AppsV1().Deployments(catalogNamespace).Get("apiserver", metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			if deployment.Status.AvailableReplicas == 0 {
+				return false, nil
+			}
+			return componentinstall.CheckForAPIs(restConfig, "v1beta1.servicecatalog.k8s.io")
+		},
+	}
+
+	return component.MakeReady(
+		c.OCImage,
+		clusterAdminKubeConfigBytes,
+		params).Install(dockerClient, logdir)
+}
+
+// TODO move to template
+func getServiceCatalogClusterRoles() []rbac.ClusterRole {
```

Comment:
@deads2k is this also dealing with https://github.com/openshift/origin/issues/19163 ?

Reply:
> @deads2k is this also dealing with #19163 ?

Wasn't trying to.  Role missing maybe?